### PR TITLE
Bedevere as GH Actions

### LINKED ISSUE #344: Move to GitHub Actions
Sometimes, the 3 "bedevere/*" bots don't run and it is **not possible to merge a PR** :-(

miss-islington created https://github.com/python/cpython/pull/26413 backport PR. I approved it. But the following 3 jobs never ran:

* bedevere/issue-number Expected — Waiting for status to be reported (Required)
* bedevere/maintenance-branch-pr Expected — Waiting for status to be reported (Required)
* bedevere/news Expected — Waiting for status to be reported (Required)

Workaround for issue-number and maintenenance branch: edit the PR title, then edit to undo your edit.

Workaround for news: add "skip news" label.

When the PR changes, the bot runs, and it becomes possible to merge the PR.

Who operates these bots? How can I debug these issues?

### PR DESCRIPTION
**Pull request by Mariatta**
Instead of getting information from webhook payload,
get them from the Actions environment.
Create fixtures for GitHub Actions environment vars.

Adjust tests for __main__.py.
Remove "ping" handler, not needed in Actions

Still in Draft.

Need help with Dockerising it or make it faster.

Closes #344

### CODE REVIEW

File: .github/workflows/bedevere.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Bedevere
+
+on:
+  pull_request:
+    types:
+      - edited
+      - opened
+      - synchronize
+      - reopened
+      - edited
```
Comment (arhadthedev) [inline .github/workflows/bedevere.yml:10]: A repetition of the first list entry.

```suggestion
```

Comment (arhadthedev) [reply .github/workflows/bedevere.yml:10]: As an alternative, the first entry may be deleted instead.

**Comment by arhadthedev** [comment]
For history: this PR resolves gh-344.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.11-slim
+
+COPY requirements.txt requirements.txt
+COPY dev-requirements.txt dev-requirements.txt
```
[the remark this answers is not in the document]
Comment (Mariatta) [reply Dockerfile:4]: I've made the changes.

File: .github/workflows/bedevere.yml (hunk 1 above)
Comment (Mariatta) [reply .github/workflows/bedevere.yml:9]: Thanks, yes I've removed the duplicated event, and added empty lines at the end of file.

**Comment by Mariatta** [comment]
I'm testing this on https://github.com/Mariatta/cpython/pull/74

**Comment by hugovk** [comment]
Sorry, I accidentally pushed a test to this branch. It didn't help so I've undone it.

File: bedevere/__main__.py [hunk 3]
```diff
@@ -10,27 +10,30 @@
 from gidgethub import aiohttp as gh_aiohttp
 from gidgethub import routing
 from gidgethub import sansio
+from gidgethub import actions
 
 from . import backport, gh_issue, close_pr, filepaths, news, stage
 
-import sentry_sdk
+# import sentry_sdk
```
Comment (Mariatta) [inline bedevere/__main__.py:17]: Can we still use sentry in GitHub Actions? Is it needed? If not we can remove it from requirements.txt

**Comment by Mariatta** [comment]
The bedevere actions seem to be working at least on my CPython fork.
Not sure if anyone else would be able to try it themselves? Example PR https://github.com/Mariatta/cpython/pull/74/files

I still find that the "actions" are slower to execute compared to having it as a Web app/ webhook event.

For example, when changing the skip issue/skip news label, the actions need to re-run and takes time before it posts the status. Whereas as a webhook event, it works faster/almost immediate.

**Pull request closed without merging**